Sources/Rendering/Core/VolumeMapper/SliceHelper.js

```diff
@@ -0,0 +1,82 @@
+import macro from 'vtk.js/Sources/macros';
+import * as vtkMath from 'vtk.js/Sources/Common/Core/Math';
+import vtkPlane from 'vtk.js/Sources/Common/DataModel/Plane';
+
+function vtkSliceHelper(publicAPI, model) {
+  model.classHierarchy.push('vtkSliceHelper');
+
+  model.clipPlane1 = vtkPlane.newInstance();
+  model.clipPlane2 = vtkPlane.newInstance();
+
+  const superClass = { ...publicAPI };
+
+  function update() {
+    const n1 = model.clipPlane1.getNormalByReference();
+    n1[0] = model.normal[0];
+    n1[1] = model.normal[1];
+    n1[2] = model.normal[2];
+
+    const n2 = model.clipPlane2.getNormalByReference();
+    n2[0] = -model.normal[0];
+    n2[1] = -model.normal[1];
+    n2[2] = -model.normal[2];
+
+    vtkMath.multiplyAccumulate(
+      model.origin,
+      model.normal,
+      -model.thickness / 2,
+      model.clipPlane1.getOriginByReference()
+    );
+
+    vtkMath.multiplyAccumulate(
+      model.origin,
+      model.normal,
+      model.thickness / 2,
+      model.clipPlane2.getOriginByReference()
+    );
+
+    model.clipPlane1.modified();
+    model.clipPlane2.modified();
+  }
+
+  const subscription = publicAPI.onModified(update);
+  publicAPI.delete = () => {
+    superClass.delete();
+    subscription.unsubscribe();
+  };
+
+  publicAPI.setMapper = (mapper) => {
+    mapper.addClippingPlane(model.clipPlane1);
+    mapper.addClippingPlane(model.clipPlane2);
+  };
+}
+
+// ----------------------------------------------------------------------------
+// Object factory
+// ----------------------------------------------------------------------------
+
+const DEFAULT_VALUES = {
+  thickness: 3,
```

Comment:
a thickness of 0 (or epsilon) seems to be a better default.

Reply:
Agreed